packages/datadog_flutter_plugin/lib/src/rum/ddrum_method_channel.dart

```diff
@@ -157,3 +173,140 @@ class DdRumMethodChannel extends DdRumPlatform {
+  final RumResourceEventMapper? resourceEventMapper;
+  final RumErrorEventMapper? errorEventMapper;
+  final RumLongTaskEventMapper? longTaskEventMapper;
+
+  final InternalLogger internalLogger;
+
+  MethodCallHandler({
+    this.viewEventMapper,
+    this.actionEventMapper,
+    this.resourceEventMapper,
+    this.errorEventMapper,
+    this.longTaskEventMapper,
+    required this.internalLogger,
+  });
+
+  Future<dynamic> handleMethodCall(MethodCall call) async {
+    switch (call.method) {
+      case 'mapViewEvent':
+        return _mapViewEvent(call);
+      case 'mapActionEvent':
+        return _mapActionEvent(call);
+      case 'mapResourceEvent':
+        return _mapResourceEvent(call);
+      case 'mapErrorEvent':
+        return _mapErrorEvent(call);
+      case 'mapLongTaskEvent':
+        return _mapLongTaskEvent(call);
+    }
+
+    throw MissingPluginException(
+        'Could not find a method to call for ${call.method}');
+  }
+
+  Map<String, Object?>? _callMapper<T>(
+    String mapperName,
+    Map encoded,
+    T? Function(T)? mapper,
+    Map<String, dynamic> Function(T) encode,
+    T Function(Map) decode,
+  ) {
+    try {
+      if (mapper == null) {
+        final st = StackTrace.current;
+        internalLogger.sendToDatadog(
+            '$mapperName called but no $mapperName is set,',
+            st,
+            'InternalDatadogError');
+        return mapperError;
+      }
+
+      final event = decode(encoded);
+
+      T? mappedEvent = event;
+      try {
+        mappedEvent = mapper(event);
+        if (mappedEvent == null) {
+          return null;
+        }
+      } catch (e) {
+        internalLogger.error(
+            '$mapperName threw an exception: ${e.toString()}.\nReturning unmapped event.');
+        return mapperError;
+      }
+
+      final mappedJson = encode(mappedEvent);
+      return mappedJson;
+    } catch (e, st) {
+      internalLogger.sendToDatadog('Error mapping view event: ${e.toString()}',
+          st, e.runtimeType.toString());
+    }
+
+    // Return a special map which will indicate to native code something went wrong, and
+    // we should send the unmodified event.
+    return mapperError;
+  }
+
+  Map<Object, Object?>? _mapViewEvent(MethodCall call) {
+    final viewEventJson = call.arguments['event'] as Map;
+    return _callMapper<RumViewEvent>(
+      'mapViewEvent',
+      viewEventJson,
+      viewEventMapper,
+      (e) => e.toJson(),
+      RumViewEvent.fromJson,
+    );
+  }
+
+  Map<Object, Object?>? _mapActionEvent(MethodCall call) {
+    final eventJson = call.arguments['event'] as Map;
+    return _callMapper<RumActionEvent>(
+      'mapActionEvent',
+      eventJson,
+      actionEventMapper,
+      (e) => e.toJson(),
+      RumActionEvent.fromJson,
+    );
+  }
+
+  Map<Object, Object?>? _mapResourceEvent(MethodCall call) {
+    final eventJson = call.arguments['event'] as Map;
+    return _callMapper<RumResourceEvent>(
+      'mapActionEvent',
```

Comment:
typo, wrong mapper is called

```suggestion
      'mapResourceEvent',
```

Reply:
:+1: nice catch. Thankfully it's just for error messages.